include/adapter/arbitrary_interface.h

```diff
@@ -0,0 +1,286 @@
+
+    // Allocate a vector containing the vertices
+    std::vector<double> received_coordinates(received_mesh_size * dim);
+    interface_nodes_ids.resize(received_mesh_size);
+
+    this->precice->getMeshVerticesWithIDs(this->mesh_id,
+                                          received_mesh_size,
+                                          interface_nodes_ids.data(),
+                                          received_coordinates.data());
+
+    // Transform the received points into a vector of points
+    std::vector<Point<dim>> received_points(received_mesh_size);
+    for (int i = 0; i < received_mesh_size; ++i)
+      for (int d = 0; d < dim; ++d)
+        {
+          AssertIndexRange(i * dim + d, received_coordinates.size());
+          received_points[i][d] = received_coordinates[i * dim + d];
+        }
+
+    // TODO: Maybe perform some coarse pre-filtering here
+    filter_vertices_to_local_partition(
+      *(this->mf_data->get_mapping_info().mapping),
+      this->mf_data->get_dof_handler(0).get_triangulation(),
+      received_points);
+
+    Assert(this->read_data_id == -1, ExcInternalError());
+    Assert(this->write_data_id != -1, ExcInternalError());
+    this->print_info(false);
+  }
+
+
+  template <int dim, typename VectorizedArrayType>
+  std::string
+  ArbitraryInterface<dim, VectorizedArrayType>::get_interface_type() const
+  {
+    return "arbitrary nodes defined by the coupling partner ";
+  }
+
+
+
+  template <int dim, typename VectorizedArrayType>
+  std::vector<
+    std::pair<typename Triangulation<dim>::active_cell_iterator, Point<dim>>>
+  ArbitraryInterface<dim, VectorizedArrayType>::
+    filter_vertices_to_local_partition(const Mapping<dim> &          mapping,
+                                       const Triangulation<dim> &    tria,
+                                       const std::vector<Point<dim>> points_in,
+                                       double tolerance) const
+  {
+    std::vector<
+      std::pair<typename Triangulation<dim>::active_cell_iterator, Point<dim>>>
+      unique_points;
+
+    GridTools::Cache<dim>                             cache(tria, mapping);
+    typename Triangulation<dim>::active_cell_iterator cell_hint;
+    const std::vector<bool>                           marked_vertices;
+
+    const unsigned int my_rank =
+      Utilities::MPI::this_mpi_process(tria.get_communicator());
+
+    for (const auto &point : points_in)
+      {
+        const auto first_cell = GridTools::find_active_cell_around_point(
+          cache, point, cell_hint, marked_vertices, tolerance);
+
+        cell_hint = first_cell.first;
+
+        if (cell_hint.state() != IteratorState::valid)
+          continue;
+
+        const auto active_cells_around_point =
+          GridTools::find_all_active_cells_around_point(
+            cache.get_mapping(),
+            cache.get_triangulation(),
+            point,
+            tolerance,
+            first_cell);
```

Comment:
@peterrum I think this concept is rather new in the library. The problem is here is the type of the `cell_iterator`, which is derived from `Triangulation::active_cell_iterator`. `FEPointEvaluation requires `iterator->get_dof_values()` which is only defined for `DoFHandler::cell_iterator`. The `MeshType` is a template parameter for some versions of ` find_active_cells_around_point`, but not for your fancy version using `GridTools::Cache`. 
Are you aware of this or is there some common derived class allowing me to access `get_dof_values()`?
(Sorry for bothering you again, last time)

Reply:
The cache only lives on the triangulation and as result it only can return a CellAccessor. But you can create you own DoFCellAccessor via:
https://github.com/kronbichler/adaflo/blob/af295a4257cbd009e3b775e6f97b5f619f3abff2/include/adaflo/sharp_interface_util.h#L360-L363

Reply:
Haha funny that it's so simple, thanks!